src/rtutils.c

```diff
@@ -777,6 +804,42 @@ static int jl_static_is_function_(jl_datatype_t *vt) JL_NOTSAFEPOINT {
     return 0;
 }
 
+static size_t jl_static_show_float(JL_STREAM *out, double v,
+                                   jl_datatype_t *vt) JL_NOTSAFEPOINT
+{
+    size_t n = 0;
+    // TODO: non-canonical NaNs do not round-trip
+    // TOOD: BFloat16
+    const char *size_suffix = vt == jl_float16_type ? "16" :
+                              vt == jl_float32_type ? "32" :
+                                                      "";
+    // Digits required to print p significand bits: ceil(p * log(10, 2))
+    //   Float16   4
+    //   Float32   8
+    //   Float64  17
+    if (isnan(v)) {
+        n += jl_printf(out, "NaN%s", size_suffix);
+    }
+    else if (isinf(v)) {
+        n += jl_printf(out, "%sInf%s", v < 0 ? "-" : "", size_suffix);
+    }
+    else if (vt == jl_float64_type) {
+        // If no decimal point or exponent will be printed, we must add ".0"
+        double ipart, fpart;
+        fpart = modf(v, &ipart);
+        int dotzero = v == 0 || (fpart == 0.0 && fabs(ipart) < 1e17);
+        n += jl_printf(out, "%.17g%s", v, dotzero ? ".0" : "");
+    }
+    else if (vt == jl_float32_type) {
+        n += jl_printf(out, "Float32(%.8g)", v);
```

Comment:
This can be implemented compactly similar to f64 case, with style '%e' and  `e` replaced with `f` in the buffer